src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Here is some cmake code that seems to do what is desired (put it before the define for CARET_DOTFCN, and then use `WORKBENCH_USE_SIMD AND CPUINFO_COMPILES` in conditionals):

```
#
# Define flag to avoid trying to compile SIMD stuff (coded for x86 only)
#
SET(WORKBENCH_USE_SIMD TRUE CACHE BOOL "try to compile with SIMD support")

#
# If we should try to use SIMD, check whether cpuinfo compiles
#
IF (WORKBENCH_USE_SIMD)
    TRY_COMPILE(CPUINFO_COMPILES ${CMAKE_CURRENT_BINARY_DIR}/cpuinfo_compile_test ${CMAKE_CURRENT_SOURCE_DIR}/kloewe/cpuinfo cpuinfo)
    MESSAGE("CPUINFO_COMPILES: ${CPUINFO_COMPILES}")
ENDIF (WORKBENCH_USE_SIMD)
```